tests/storages_tests/test_storages.py

```diff
@@ -267,6 +267,50 @@ def test_create_new_trial_id(storage_init_func):
     assert trials[0].system_attrs == {'_number': 0}
 
 
+@parametrize_storage
+def test_create_new_trial_id_with_template_trial(storage_init_func):
+    # type: (Callable[[], BaseStorage]) -> None
+
+    storage = storage_init_func()
+
+    now = datetime.now()
+    template_trial = FrozenTrial(
+        state=TrialState.COMPLETE,
+        value=10000,
+        datetime_start=now,
+        datetime_complete=now,
+        params={'x': 0.5},
+        distributions={'x': UniformDistribution(0, 1)},
+        user_attrs={'foo': 'bar'},
+        system_attrs={'baz': 123},
+        intermediate_values={1: 10, 2: 100, 3: 1000},
+
+        number=-1,  # dummy value (unused)
+        trial_id=-1,  # dummy, value (unused)
+    )
+
+    study_id = storage.create_new_study_id()
+    trial_id = storage.create_new_trial_id(study_id, template_trial=template_trial)
+
+    trials = storage.get_all_trials(study_id)
+    assert len(trials) == 1
+    assert trials[0].trial_id == trial_id
+    assert trials[0].number == 0
+    assert trials[0].state == template_trial.state
+    assert trials[0].state == template_trial.state
```

Comment:
This assertion is duplicated.

Reply:
Removed: https://github.com/pfnet/optuna/pull/464/commits/5c310dc7639f4cb7b4c1df439cd3c5bf2eae3a2e
Thank you for pointing it out.